pytorch_lightning/callbacks/pruning.py

```diff
@@ -295,96 +278,62 @@ def apply_pruning(self, trainer: 'pl.Trainer', pl_module: LightningModule):
 
     def on_before_accelerator_backend_setup(self, trainer, pl_module):
         parameters_to_prune = self.sanitize_parameters_to_prune(
-            pl_module, self._parameters_to_prune, parameters=self._parameter_names)
+            pl_module, self._parameters_to_prune, parameters=self._parameter_names
+        )
 
         self._parameters_to_prune = self.filter_parameters_to_prune(parameters_to_prune)
 
         if self._use_lottery_ticket_hypothesis:
-            # make a copy of copy of orginal weights.
+            # make a copy of copy of original weights.
             self._initial_parameters_to_prune = [(deepcopy(m), n) for m, n in self._parameters_to_prune]
 
     def on_epoch_end(self, trainer, pl_module):
-        self.apply_pruning(trainer, pl_module)
+        self.apply_pruning(trainer.current_epoch)
 
         if self.make_pruning_permanent:
             self._make_pruning_permanent()
 
-    @staticmethod
-    def _sanitize_parameters_to_prune(p):
-        """
-        Check the provide element is a pair with:
-            * nn.Module
-            * str
-
-        Example::
-
-            parameters_to_prune = [
-                (model.mlp_1, "weight"),
-                (model.mlp_2, "weight")
-            ]
-        """
-        return len(p) == 2 and isinstance(p[0], nn.Module) and isinstance(p[1], str)
-
     @staticmethod
     def sanitize_parameters_to_prune(
         pl_module: LightningModule,
-        parameters_to_prune: Optional[List[Tuple[nn.Module, str]]],
-        parameters: List[str] = ["weight"]
-    ) -> List:
+        parameters_to_prune: Optional[List[Tuple[nn.Module, str]]] = None,
+        parameters: Optional[List[str]] = None,
+    ) -> List[Tuple[nn.Module, str]]:
         """
-        This function is responsible to check provided `parameters_to_prune` and `parameters`.
+        This function is responsible to check provided ``parameters_to_prune` and `parameters`.
         If parameters_to_prune is None, parameters_to_prune will be generated from all parameters of the model.
         """
+        parameters = parameters or ModelPruning.PARAMETER_NAMES
 
-        is_parameters_to_prune_none = parameters_to_prune is None
         current_modules = [
-            m for m in pl_module.modules()
-            if not isinstance(m, (LightningModule, ModuleDict, ModuleList))
+            m for m in pl_module.modules() if not isinstance(m, (LightningModule, ModuleDict, ModuleList))
```

Comment:
Forgot Sequential there.

Reply:
Let's move this to _MODULE_CONTAINERS = (LightningModule, ModuleDict, ModuleList, Sequential)